aws/aws-kinesis-firehose/main.tf

```diff
@@ -0,0 +1,72 @@
+locals {
+  destinations = {
+    "s3" = "extended_s3"
+  }
+
+  default_key_prefix     = "data/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
+  partitioned_key_prefix = "data/!{partitionKeyFromQuery:partition_key}/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
+
+  key_prefix = var.dynamic_partitioning_enabled ? local.partitioned_key_prefix : local.default_key_prefix
+  name       = "${var.name}-firehose"
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "main" {
+  name        = local.name
+  destination = local.destinations[var.destination]
+
+  dynamic "extended_s3_configuration" {
+    for_each = var.dynamic_partitioning_enabled ? [1] : []
+    content {
+      role_arn            = aws_iam_role.firehose_role.arn
+      bucket_arn          = var.bucket_arn
+      prefix              = local.key_prefix
+      error_output_prefix = "errors/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/!{firehose:error-output-type}/"
+      buffer_size         = var.buffer_size     # In MB
```

Comment:
nit: for these params in the massdriver.yaml we've been adding the unit as a suffix. `buffer_size_mb`. What are your thoughts on doing that here?

Reply:
Fan of that over comments